src/openloco/windows/toolbar_top.cpp

```diff
@@ -0,0 +1,1236 @@
+                break;
+
+            case widx::road_menu:
+                road_menu_mouse_down(window, widgetIndex);
+                break;
+
+            case widx::port_menu:
+                port_menu_mouse_down(window, widgetIndex);
+                break;
+
+            case widx::build_vehicles_menu:
+                build_vehicles_menu_mouse_down(window, widgetIndex);
+                break;
+
+            case widx::vehicles_menu:
+                vehicles_menu_mouse_down(window, widgetIndex);
+                break;
+
+            case widx::stations_menu:
+                stations_menu_mouse_down(window, widgetIndex);
+                break;
+
+            case widx::towns_menu:
+                towns_menu_mouse_down(window, widgetIndex);
+                break;
+        }
+    }
+
+    static void on_dropdown(window* window, widget_index widgetIndex, int16_t itemIndex)
+    {
+        registers regs;
+        regs.esi = (int32_t)window;
+        regs.edx = widgetIndex;
+        regs.ax = itemIndex;
+
+        switch (widgetIndex)
+        {
+            case widx::loadsave_menu:
+                loadsave_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::audio_menu:
+                audio_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::zoom_menu:
+                zoom_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::rotate_menu:
+                rotate_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::view_menu:
+                view_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::terraform_menu:
+                terraform_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::railroad_menu:
+                railroad_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::road_menu:
+                road_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::port_menu:
+                port_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::build_vehicles_menu:
+                build_vehicles_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::vehicles_menu:
+                vehicles_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::stations_menu:
+                stations_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+
+            case widx::towns_menu:
+                towns_menu_dropdown(window, widgetIndex, itemIndex);
+                break;
+        }
+    }
+
+    static void on_update(window* window)
+    {
+        loco_global<int32_t, 0x9C86F8> _9C86F8;
```

Comment:
If we remove the logic regarding `0x9C86F8`, we should probably get rid of this as well

Reply:
Good call.